src/storage/src/source/mysql/replication.rs

```diff
@@ -353,15 +353,9 @@ pub(crate) fn render<G: Scope<Timestamp = GtidPartition>>(
 
                         events::handle_rows_event(data, &mut repl_context, new_gtid).await?;
 
-                        // Advance the frontier up to the point right before this GTID
-                        // We are being careful here and not advancing beyond this GTID since we aren't
-                        // sure that other mysql event-types may need to be supported that present other
-                        // data to commit at this timestamp. In the future if we are sure that this RowsEvent
-                        // represents all the data that we need to commit at this timestamp and that there is no
-                        // then we can do so (like we currently do when handling Heartbeat Events).
-                        // This is still useful to allow data committed at previous timestamps to become available
-                        // without waiting for the next break in data at which point we should receive a Heartbeat
-                        // Event.
+                        // Advance the frontier up to the point right before this GTID, since we
+                        // might still see other events that are part of this same GTID, such as
+                        // row events for multiple tables or large row events split into multiple.
```

Comment:
during debugging I saw that mysql sends more than one RowsEvent for large transactions. This means that we're going to have to live with this other issue https://github.com/MaterializeInc/materialize/issues/24630 unless we can find some other way to know if a GTID is 'complete'